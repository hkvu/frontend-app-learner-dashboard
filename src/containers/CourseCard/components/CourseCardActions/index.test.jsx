import { shallow } from 'enzyme';

import { reduxHooks } from 'hooks';

import CourseCardActions from '.';

jest.mock('hooks', () => ({
  reduxHooks: {
    useCardCourseRunData: jest.fn(),
    useCardEnrollmentData: jest.fn(),
    useCardEntitlementData: jest.fn(),
  },
}));

jest.mock('./SelectSessionButton', () => 'SelectSessionButton');
jest.mock('./ViewCourseButton', () => 'ViewCourseButton');
jest.mock('./BeginCourseButton', () => 'BeginCourseButton');
jest.mock('./ResumeButton', () => 'ResumeButton');

describe('CourseCardActions', () => {
  const props = {
    cardId: 'cardId',
  };
  const createWrapper = ({
    isEntitlement, isFulfilled, isArchived, isVerified, hasStarted,
  }) => {
    reduxHooks.useCardEntitlementData.mockReturnValueOnce({ isEntitlement, isFulfilled });
    reduxHooks.useCardCourseRunData.mockReturnValueOnce({ isArchived });
    reduxHooks.useCardEnrollmentData.mockReturnValueOnce({ isVerified, hasStarted });
    return shallow(<CourseCardActions {...props} />);
  };
  describe('snapshot', () => {
    test('show upgrade button when not verified and not entitlement', () => {
      const wrapper = createWrapper({
        isEntitlement: false, isFulfilled: false, isArchived: false, isVerified: false, hasStarted: false,
      });
      expect(wrapper).toMatchSnapshot();
    });
    test('show select session button when not verified and entitlement', () => {
      const wrapper = createWrapper({
        isEntitlement: true, isFulfilled: false, isArchived: false, isVerified: false, hasStarted: false,
      });
      expect(wrapper).toMatchSnapshot();
    });
    test('show begin course button when verified and not entitlement and has started', () => {
      const wrapper = createWrapper({
        isEntitlement: false, isFulfilled: false, isArchived: false, isVerified: true, hasStarted: false,
      });
      expect(wrapper).toMatchSnapshot();
    });
    test('show resume button when verified and not entitlement and has started', () => {
      const wrapper = createWrapper({
        isEntitlement: false, isFulfilled: false, isArchived: false, isVerified: true, hasStarted: true,
      });
      expect(wrapper).toMatchSnapshot();
    });
    test('show view course button when not verified and entitlement and fulfilled', () => {
      const wrapper = createWrapper({
        isEntitlement: true, isFulfilled: true, isArchived: false, isVerified: false, hasStarted: false,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    it('show select session button when not verified and entitlement', () => {
      const wrapper = createWrapper({
        isEntitlement: true, isFulfilled: false, isArchived: false, isVerified: false, hasStarted: false,
      });
      expect(wrapper.find('SelectSessionButton')).toHaveLength(1);
    });
    it('show begin course button when verified and not entitlement and has started', () => {
      const wrapper = createWrapper({
        isEntitlement: false, isFulfilled: false, isArchived: false, isVerified: true, hasStarted: false,
      });
      expect(wrapper.find('BeginCourseButton')).toHaveLength(1);
    });
    it('show resume button when verified and not entitlement and has started', () => {
      const wrapper = createWrapper({
        isEntitlement: false, isFulfilled: false, isArchived: false, isVerified: true, hasStarted: true,
      });
      expect(wrapper.find('ResumeButton')).toHaveLength(1);
    });
    it('show view course button when not verified and entitlement and fulfilled', () => {
      const wrapper = createWrapper({
        isEntitlement: true, isFulfilled: true, isArchived: false, isVerified: false, hasStarted: false,
      });
      expect(wrapper.find('ViewCourseButton')).toHaveLength(1);
    });
    it('show view course button when not verified and entitlement and fulfilled and archived', () => {
      const wrapper = createWrapper({
        isEntitlement: true, isFulfilled: true, isArchived: true, isVerified: false, hasStarted: false,
      });
      expect(wrapper.find('ViewCourseButton')).toHaveLength(1);
    });
  });
});
