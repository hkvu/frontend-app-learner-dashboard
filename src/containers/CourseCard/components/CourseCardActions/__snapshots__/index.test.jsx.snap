// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCardActions snapshot show begin course button when verified and not entitlement and has started 1`] = `
<ActionRow
  data-test-id="CourseCardActions"
>
  <BeginCourseButton
    cardId="cardId"
  />
</ActionRow>
`;

exports[`CourseCardActions snapshot show resume button when verified and not entitlement and has started 1`] = `
<ActionRow
  data-test-id="CourseCardActions"
>
  <ResumeButton
    cardId="cardId"
  />
</ActionRow>
`;

exports[`CourseCardActions snapshot show select session button when not verified and entitlement 1`] = `
<ActionRow
  data-test-id="CourseCardActions"
>
  <SelectSessionButton
    cardId="cardId"
  />
</ActionRow>
`;

exports[`CourseCardActions snapshot show upgrade button when not verified and not entitlement 1`] = `
<ActionRow
  data-test-id="CourseCardActions"
>
  <BeginCourseButton
    cardId="cardId"
  />
</ActionRow>
`;

exports[`CourseCardActions snapshot show view course button when not verified and entitlement and fulfilled 1`] = `
<ActionRow
  data-test-id="CourseCardActions"
>
  <ViewCourseButton
    cardId="cardId"
  />
</ActionRow>
`;
